contracts/registry/implementation/IdentityRegistryStorageERC3643.sol

```diff
@@ -0,0 +1,173 @@
+import "@onchain-id/solidity/contracts/factory/IIdFactory.sol";
+import "@onchain-id/solidity/contracts/interface/IIdentity.sol";
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+
+/// Errors
+
+/// @dev Thrown when address is already stored
+error AddressAlreadyStored();
+
+/// @dev Thrown when address is not yet stored.
+error AddressNotYetStored();
+
+/// @dev Thrown when trying to call write functions.
+error ContractIsReadOnly();
+
+/// @dev Thrown when maximum numbe of identity registry by identity registry storage is reached.
+/// @param _max miximum number of IR by IRS.
+error MaxIRByIRSReached(uint256 _max);
+
+
+contract IdentityRegistryStorage is IERC3643IdentityRegistryStorage, AgentRoleUpgradeable, IRSStorage, IERC165 {
+    IIdFactory internal _iidFactory;
+    function init(_identityFactoryAddress) external initializer {
+        require(_identityFactoryAddress != address(0), ZeroAddress());
+        _iidFactory = IIdFactory(_identityFactoryAddress);
+        __Ownable_init();
+    }
+
+    /**
+     *  @dev See {IIdentityRegistryStorage-addIdentityToStorage}.
+     */
+    function addIdentityToStorage() external view override onlyAgent {
+        revert ContractIsReadOnly();
+    }
+
+    /**
+     *  @dev See {IIdentityRegistryStorage-modifyStoredIdentity}.
+     */
+    function modifyStoredIdentity() external view override onlyAgent {
+        revert ContractIsReadOnly();
+
+    }
+
+    /**
+     *  @dev See {IIdentityRegistryStorage-modifyStoredInvestorCountry}.
+     */
+    function modifyStoredInvestorCountry() external view override onlyAgent {
+        revert ContractIsReadOnly();
+
+    }
+
+    /**
+     *  @dev See {IIdentityRegistryStorage-removeIdentityFromStorage}.
+     */
+    function removeIdentityFromStorage() external view override onlyAgent {
+        revert ContractIsReadOnly();
+    }
+
+    /**
+     *  @dev See {IIdentityRegistryStorage-bindIdentityRegistry}.
+     */
+    function bindIdentityRegistry() external view override {
+        revert ContractIsReadOnly();
+    }
+
+    /**
+     *  @dev See {IIdentityRegistryStorage-unbindIdentityRegistry}.
+     */
+    function unbindIdentityRegistry() external view override {
+        revert ContractIsReadOnly();
+    }
+
+    /**
+     *  @dev See {IIdentityRegistryStorage-linkedIdentityRegistries}.
+     */
+    function linkedIdentityRegistries() external view override returns (address[] memory) {
+        return _identityRegistries;
+    }
+
+    /**
+     *  @dev See {IIdentityRegistryStorage-storedIdentity}.
+     */
+    function storedIdentity(address _userAddress) external view override returns (IIdentity) {
+        return IIdentity(_iidFactory.getIdentity(_userAddress));
+    }
+
+    /**
+     *  @dev See {IIdentityRegistryStorage-storedInvestorCountry}.
+     */
+    function storedInvestorCountry(address _userAddress) external view override returns (uint16) {
+        return _identities[_userAddress].investorCountry;
```

Comment:
just return 42 or something, by default, in case the `_identities[_userAddress].investorCountry` value is 0 
Should have a way for OID owners to publish their country code there in a decentralized way, so that a wallet that is linked to an OID on the IdFactory can update their own country code on this contract